eventhandling/getSliEvent.go

```diff
@@ -0,0 +1,275 @@
+	log.Println("Checking for custom SLI queries")
+
+	customQueries, err := keptnHandler.GetSLIConfiguration(project, stage, service, utils.SliResourceURI)
+	if err != nil {
+		return nil, err
+	}
+
+	return customQueries, nil
+}
+
+// getPrometheusAPIURL fetches the prometheus API URL for the provided project (e.g., from Kubernetes configmap)
+func getPrometheusAPIURL(project string, kubeClient v1.CoreV1Interface) (string, error) {
+	log.Println("Checking if external prometheus instance has been defined for project " + project)
+	secret, err := kubeClient.Secrets(env.PodNamespace).Get("prometheus-credentials-"+project, metav1.GetOptions{})
+
+	// return cluster-internal prometheus URL if no secret has been found
+	if err != nil {
+		log.Println("could not retrieve or read secret: " + err.Error())
+		log.Println("No external prometheus instance defined for project " + project + ". Using default: " + env.PrometheusEndpoint)
+		return env.PrometheusEndpoint, nil
+	}
+
+	pc := &prometheusCredentials{}
+	err = yaml.Unmarshal(secret.Data["prometheus-credentials"], pc)
+
+	if err != nil {
+		log.Println("Could not parse credentials for external prometheus instance: " + err.Error())
+		return "", errors.New("invalid credentials format found in secret 'prometheus-credentials-" + project)
+	}
+	log.Println("Using external prometheus instance for project " + project + ": " + pc.URL)
+	prometheusURL := generatePrometheusURL(pc)
+
+	return prometheusURL, nil
+}
+
+func generatePrometheusURL(pc *prometheusCredentials) string {
+	prometheusURL := pc.URL
+
+	credentialsString := ""
+
+	if pc.User != "" && pc.Password != "" {
+		credentialsString = url.QueryEscape(pc.User) + ":" + url.QueryEscape(pc.Password) + "@"
+	}
+	if strings.HasPrefix(prometheusURL, "https://") {
+		prometheusURL = strings.TrimPrefix(prometheusURL, "https://")
+		prometheusURL = "https://" + credentialsString + prometheusURL
+	} else if strings.HasPrefix(prometheusURL, "http://") {
+		prometheusURL = strings.TrimPrefix(prometheusURL, "http://")
+		prometheusURL = "http://" + credentialsString + prometheusURL
+	} else {
+		// assume https transport
+		prometheusURL = "https://" + credentialsString + prometheusURL
+	}
+	return strings.Replace(prometheusURL, " ", "", -1)
+}
+
+func sendGetSLIStartedEvent(inputEvent cloudevents.Event, eventData *keptnv2.GetSLITriggeredEventData, keptnContext interface{}) error {
+
+	source, _ := url.Parse(utils.ServiceName)
+
+	getSLIStartedEvent := keptnv2.GetSLIStartedEventData{
+		EventData: keptnv2.EventData{
+			Project: eventData.Project,
+			Stage:   eventData.Stage,
+			Service: eventData.Service,
+			Labels:  eventData.Labels,
+			Status:  keptnv2.StatusSucceeded,
+			Result:  keptnv2.ResultPass,
+		},
+	}
+
+	event := cloudevents.NewEvent()
+	event.SetType(keptnv2.GetStartedEventType(keptnv2.GetSLITaskName))
+	event.SetSource(source.String())
```

Comment:
Why do we parse the ServiceName URL to call `.String()` right after?
same thing in line 229

Reply:
This is copied from the original source in main.go, and will be refactored later